staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor.go

```diff
@@ -107,7 +88,57 @@ func (c *convertor) ConvertToTable(ctx genericapirequest.Context, obj runtime.Ob
 		cells[0] = name
 		cells[1] = age
 		for _, column := range c.columns {
-			if err := column.Execute(buf, obj); err != nil {
+			results, err := column.FindResults(obj.(runtime.Unstructured).UnstructuredContent())
+			if err != nil || len(results) == 0 {
+				cells = append(cells, nil)
+				continue
+			}
+			// as we only support simple JSON path, we can assume to have only one result
+			result := results[0]
+
+			if len(result) == 1 {
+				// try to preserve JSON value type, only convert to string as fallback (= fall through)
+				cell := result[0].Interface()
+				switch typed := result[0].Interface().(type) {
+				case bool, int64, float64, string: // all primitive types returned by json.Unmarshal
+					cells = append(cells, cell)
+					continue
+				case []interface{}:
+					primitiveValues := true
+					for _, x := range typed {
+						switch x.(type) {
+						case bool, int64, float64, string: // all primitive types returned by json.Unmarshal
+						default:
+							primitiveValues = false
+							break
+						}
+					}
+					if primitiveValues {
+						cells = append(cells, cell)
+						continue
+					}
+					// TODO: types.go for Cells mentiones "simple maps", but resttest.go forbid them. Clarify!
```

Comment:
Simple maps was intended for things like labels, but we haven't gotten there yet.  I think the important outcome is that clients *must* hide/handle types they don't recognize, but we have to realize that not all clients will do that.

Reply:
I have restricted the types for CRDs via validation to the primitive types for now: string, bool, integer, number.